### PR TITLE
Implements `/v0/me/capabilities`, `upholdInsufficientCapabilitiesModal`.

### PR DESCRIPTION
**Pull request by szilardszaloki**
Resolves https://github.com/brave/brave-browser/issues/22508.

Implements `/v0/me/capabilities` and `upholdInsufficientCapabilitiesModal`.
Removes:
- Uphold user status inspection
- explicit CDD check
- `upholdBlockedUserModal`, `upholdPendingUserModal`, `upholdRestrictedUserModal`, `upholdCustomerDueDiligenceRequiredModal`

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

The `Test Plan` of https://github.com/brave/brave-core/pull/11857 can be applied here. Make sure to edit the response of `/v0/me/capabilities` with Charles Proxy so that either `receives` or `sends` is set to `false`.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc [hunk 1]
```diff
@@ -142,20 +114,54 @@ void UpholdWallet::OnGetUser(const type::Result result,
...
+    BLOG(0, "Access token expired!");
     // Entering NOT_CONNECTED or DISCONNECTED_VERIFIED.
-    ledger_->uphold()->DisconnectWallet("");
+    ledger_->uphold()->DisconnectWallet(
+        ledger::notifications::kWalletDisconnected);
+    return callback(type::Result::EXPIRED_TOKEN);
+  }
+
+  if (result != type::Result::LEDGER_OK || capabilities.empty() ||
+      !capabilities.count("receives") || !capabilities.count("sends")) {
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc:146]: By using string keys here, it seems that this class is taking on some of the "parsing" responsibilities associated with the endpoint code.

Another option might be to have the `GetCapabilities` endpoint class return a structure instead, that contains the information that callers care about (maybe something with `can_send` and `can_receive` bools).

Comment (szilardszaloki) [reply vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_wallet.cc]: @zenparsing Yeah, good point — amended in https://github.com/brave/brave-core/pull/13474/commits/4eba72eaa123b8882409d9d7b4331ab9182512b0.

**Review by zenparsing — APPROVED** [review]

**Review by emerick — APPROVED** [review]
LGTM

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave | 1.41.30 Chromium: 102.0.5005.78 (Official Build) nightly (x86_64)
-- | --
Revision | df6dbb5a9fd82af3f567198af2eb5fb4876ef99c-refs/branch-heads/5005_59@{#3}
OS | macOS Version 12.4 (Build 21F79)
```

Verified test plan from https://github.com/brave/brave-core/pull/13474#issue-1248760790 for the following:

<details>
<summary>Case 1: Clean profile, "sends" capability is false - InProgress</summary>

</details>

<details>
<summary>Case 2: Upgrade profile, "sends" capability is false, brave://rewards page is open and active after upgrade - TBD</summary>

</details>

<details>
<summary>Case 3: Upgrade profile, "sends" capability is false, brave://rewards page is not open after upgrade - TBD</summary>

</details>

<details>
<summary>Case 4: Clean profile, "receives" capability is false - TBD</summary>

</details>

<details>
<summary>Case 5: Upgrade profile, "receives" capability is false, brave://rewards page is open and active after upgrade - TBD</summary>

</details>

<details>
<summary>Case 6: Upgrade profile, "receives" capability is false, brave://rewards page is not open after upgrade - TBD</summary>

</details>

Also verified the following cases for an account in good standing/full capabilities:

<details>
<summary>Case 7: Clean profile, both "sends" and "receives" are true - TBD</summary>

</details>

<details>
<summary>Case 8: Upgrade profile, "sends" and "receives" are true, brave://rewards page is open and active after upgrade - TBD</summary>

</details>

<details>
<summary>Case 9: Upgrade profile, "sends" and "receives" are true, brave://rewards page is not open after upgrade - TBD</summary>

</details>

**Pull request merged**